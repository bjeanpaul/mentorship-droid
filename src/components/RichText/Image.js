import React, { PropTypes } from 'react';
import { View, Image as RNImage } from 'react-native';

import { ImageUrl } from 'src/api';


const Image = ({
  url,
  styles = {},
}) => (
  <View style={styles.mentorshipImageContainer}>
    <RNImage
      style={styles.mentorshipImage}
      source={url.resize(328, 183).toSource()}
    />
  </View>
);


Image.propTypes = {
  styles: PropTypes.object,
  url: PropTypes.instanceOf(ImageUrl).isRequired,
};


export default Image;
