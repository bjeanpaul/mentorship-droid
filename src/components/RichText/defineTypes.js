import pascalCase from 'to-pascal-case';
import { assign, mapKeys, mapValues } from 'lodash';
import React, { PropTypes } from 'react';
import richText, { richTextTypes } from 'src/richText';


const defineType = (type, RichText) => assign(({ content }) => (
  <RichText content={richText(type)(content)} />
), {
  propTypes: { content: PropTypes.any.isRequired },
});


const defineTypes = RichText => {
  const types = mapKeys(richTextTypes, (_, name) => pascalCase(name));
  return mapValues(types, type => defineType(type, RichText));
};


export default defineTypes;
