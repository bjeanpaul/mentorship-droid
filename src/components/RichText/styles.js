import { omit } from 'lodash';
import mdStyles from 'react-native-simple-markdown/styles';

import { FONT, FONT_WEIGHT } from 'src/constants/styles';
import { DEVICE_WIDTH } from 'src/constants/styles';
import colors from 'src/constants/colors';
import Text from 'src/components/Text';


const FONT_SIZE_NORMAL = 18;


export default {
  container: {
    paddingHorizontal: 16,
  },
  mentorshipImageContainer: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 20,
  },
  mentorshipImage: {
    width: 328,
    height: 183,
    maxWidth: DEVICE_WIDTH - 20,
  },
  del: {
    textDecorationLine: 'line-through',
  },
  blockQuoteSection: {
    ...mdStyles.blockQuoteSection,
    marginBottom: 20,
  },
  blockQuoteSectionBar: {
    ...mdStyles.blockQuoteSectionBar,
    backgroundColor: colors.BG_DARK,
  },
  text: Text.styles.default,
  paragraph: [Text.types.paragraph, {
    fontSize: FONT_SIZE_NORMAL,
  }],
  u: {
  },
  list: {
    marginBottom: 20,
  },
  listItemText: {
    ...omit(mdStyles.listItemText, 'color'),
    flex: 1,
    fontSize: FONT_SIZE_NORMAL,
  },
  listItemNumber: {
    fontSize: FONT_SIZE_NORMAL,
    color: colors.TEXT_DEFAULT,
  },
  listItemBullet: {
    fontSize: FONT_SIZE_NORMAL,
    color: colors.TEXT_DEFAULT,
  },
  heading: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
  },
  heading1: {
    fontSize: 24,
    fontFamily: FONT.MEDIUM,
    fontWeight: FONT_WEIGHT.MEDIUM,
    textAlign: 'center',
    marginTop: 11,
    marginBottom: 25,
  },
  heading2: {
    fontSize: 18,
    marginTop: 27,
    marginBottom: 13,
  },
  heading3: {
    fontSize: 16,
    marginTop: 20,
    marginBottom: 10,
  },
  heading4: {
    fontSize: 15,
    marginTop: 14,
    marginBottom: 7,
  },
  heading5: {
    fontSize: 14,
    marginTop: 8,
    marginBottom: 4,
  },
  heading6: {
    fontSize: 13,
    marginTop: 6,
    marginBottom: 3,
  },
  table: {
  },
  tableHeader: {
  },
  tableRow: {
  },
};
