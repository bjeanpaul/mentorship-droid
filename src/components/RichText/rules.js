import React from 'react';
import { Text } from 'react-native';

import Image from './Image';


const rules = styles => ({
  mentorshipImage: {
    react: node => (
      <Image styles={styles} url={node.url} />
    ),
  },
  u: {
    react: (node, output, state) => {
      const children = output(node.content, {
        ...state,
        withinText: true,
      });

      return (
        <Text key={state.key} style={styles.u}>{children}</Text>
      );
    },
  },
});


export default rules;
