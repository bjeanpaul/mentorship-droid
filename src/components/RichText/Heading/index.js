import React, { PropTypes } from 'react';
import { Text } from 'react-native';

import styles from './styles';


const Heading = ({
  value,
}) => (
  <Text style={styles.default}>{value}</Text>
);


Heading.propTypes = {
  value: PropTypes.string.isRequired,
};


export default Heading;
