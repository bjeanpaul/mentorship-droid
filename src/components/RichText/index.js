import React, { PropTypes } from 'react';
import { View } from 'react-native';
import { delegate } from 'src/helpers';
import Heading from './Heading';


const renderChild = delegate('type', [
  ['heading', Heading],
]);


const RichText = ({
  content,
}) => (
  <View>
    {content.map((child, i) => <View key={i}>{renderChild(child)}</View>)}
  </View>
);


RichText.propTypes = {
  content: PropTypes.array.isRequired,
};


export default RichText;
