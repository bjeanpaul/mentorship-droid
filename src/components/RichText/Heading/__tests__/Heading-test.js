import React from 'react';
import RichText from 'src/components/RichText';


describe('RichText/Heading', () => {
  it('should render', () => {
    expect(render(
      <RichText
        content={[{
          type: 'heading',
          value: 'Foo',
        }]}
      />
    ).toJSON()).toMatchSnapshot();
  });
});
