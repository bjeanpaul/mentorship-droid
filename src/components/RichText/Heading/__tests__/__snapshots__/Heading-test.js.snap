exports[`RichText/Heading should render 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}>
      Foo
    </Text>
  </View>
</View>
`;
